### PR TITLE
add task scheduler utility

### PR DESCRIPTION
**Pull request by thrau**
This PR adds a scheduler utility that can be used to schedule tasks in an event-loop based manner, potentially without threads, which can be helpful to implement things like alarm evaluations in cloudwatch.

### CODE REVIEW

**Comment by thrau** [comment]
i'll put in a bit more work into pydoc, but at least you can see whether this is useful @steffyP

File: localstack/utils/scheduler.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        if self.fixed_rate:
+            self.deadline = self.deadline + self.period
+        else:
+            self.deadline = time.time() + self.period
+
+    def cancel(self):
+        self._cancelled = True
+
+    def run(self):
+        """
```
Comment (whummer) [inline localstack/utils/scheduler.py:57]: nit: empty doc string could be removed (or updated).

File: localstack/utils/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        Schedules a given task (function call).
+
+        :param func: the task to schedule
+        :param period: the period in which to run the task (in seconds). if not set, task will run once
+        :param fixed_rate: whether the to run at a fixed rate (neglecting execution duration of the task)
+        :param start: start time
+        :param on_error: error callback
+        :param args: additional positional arguments to pass to the function
+        :param kwargs: additional keyword arguments to pass to the function
+        :return a ScheduledTask instance
```
Comment (whummer) [inline localstack/utils/scheduler.py:108]:
```suggestion
        :return: a ScheduledTask instance
```

File: localstack/utils/scheduler.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            if self.on_error:
+                self.on_error(e)
+
+
+class Scheduler:
+    """
+    An event-loop based task scheduler that can be parallelized with an executor.
+    """
+
+    POISON = (-1, "__POISON__")
```
Comment (whummer) [inline localstack/utils/scheduler.py:73]: Love how we're introducing more [shutdown pills](https://github.com/localstack/localstack/blob/0b9f90a7221c3bfb76c4fd0c38531877ab968605/localstack/services/awslambda/invocation/version_manager.py#L95) and poison pills in the codebase.. 😄 👍

**Review by whummer — APPROVED** [review]
Looks great @thrau , very useful utility class! Over time, we could think about unifying it with the EventBridge job scheduler 👍  (https://github.com/localstack/localstack/blob/3e7050aeb3c1d2e4ccdb0213992154a22bea0d49/localstack/services/events/scheduler.py#L35), which is currently used in a few different places.

**Review by steffyP — APPROVED** [review]
This seems to work very smoothly with the metric-alarms 👍 🎉
I think the Scheduler has everything we need 😄 

@thrau: Should we merge it to master then, or should I include it in the PR for the metric-alarms?

**Pull request merged**